js/views/file_input_view.js

```diff
@@ -192,11 +192,8 @@
     /* jshint ignore:start */
     getFiles() {
       const files = this.file ? [this.file] : Array.from(this.$input.prop('files'));
-      const promises = Promise.all(
-        files.map(file => this.getFile(file))
-      );
       this.clearForm();
-      return promises;
+      return Promise.all(files.map(file => this.getFile(file)));
```

Comment:
What does `clearForm()` do? Can we call `getFile()` after it?

Reply:
Good call. Reviewing the code of `clearForm` suggests there is no issue and testing confirms, but I restored the original order with renamed `promises` --> `promise` for clarity since it makes more sense to get a file before clearing the form.